contracts/MarketGreedy.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.15;
+pragma abicoder v2;
+import "./Market.sol";
+
+abstract contract MarketGreedy is Market {
+    // MarketID => amount of cover taken for this market
+	mapping(bytes16 => uint256) private _marketCover;
+	uint256 internal _totalCover;
+	mapping(bytes16 => bytes16) internal _mostExpensivePropositionId;
+
+    function getMarketCover(bytes16 marketId) external view returns (uint256) {
+		return _marketCover[marketId];
+	}
+
+	//TODO: Should return the dedection to total exposure to be 
+	function _payout(uint256 index, bool result) internal override {
+		require(
+			_bets[index].payoutDate < block.timestamp,
+			"_settle: Payout date not reached"
+		);
+		address underlying = _vault.asset();
+
+		//_totalInPlay -= _bets[index].amount;
+		if (result == true) {
+			// Send the payout to the bet owner
+			IERC20(underlying).transfer(_bets[index].owner, _bets[index].payout);
+			_totalCover -= _bets[index].payout - _bets[index].amount;  
+		} else {
+			// Send the bet amount to the vault
+			IERC20(underlying).transfer(address(_vault), _bets[index].amount);	
+		}
+
+		//If paying out the most expensive proposition,
+		if (isMostExpensiveProposition(_bets[index].propositionId, _bets[index].marketId)) {
```

Comment:
```suggestion
		if (_isMostExpensiveProposition(_bets[index].propositionId, _bets[index].marketId)) {
```

Reply:
We should standardise here.  In other spots we have the syntax if function == true